src/components/periodic-report/periodic-report.repository.ts

```diff
@@ -74,21 +73,23 @@ export class PeriodicReportRepository extends DtoRepository(IPeriodicReport) {
       .run();
   }
 
-  async readOne(id: ID, session: Session) {
+  async readOne(id: ID) {
     const query = this.db
       .query()
-      .apply(matchRequestingUser(session))
       .match([node('node', 'PeriodicReport', { id })])
-      .apply(matchPropList)
-      .optionalMatch([
-        node('node'),
-        relation('out', '', 'reportFile', { active: true }),
-        node('reportFile', 'File'),
-      ])
-      .return('node, propList')
-      .asResult<StandardReadResult<DbPropsOfDto<PeriodicReport>>>();
+      .apply(this.hydrate())
```

Comment:
I like the hydration. This is the same functionality that would replace calling a bunch of read ones for the list queries?

Reply:
Yep exactly. That's WIP in a branch somewhere 😄

Reply:
Excellent.

Reply:
I'm going to stop reviewing. I'm starting to nitpick. It looks right to me. Tested it out too.